File: 02_week_01_c/02_mario-less/mario.c
#include <cs50.h>
#include <stdio.h>

// initiating main function
int main(void)
{
    // set height as a variable that stores an integer value
    int height;

    // start a do-while loop to make sure that the user provides an input from 1 to 8
    do
    {
        // ask the user how high the pyramids should be
        height = get_int("Height: ");
        // make sure that the input is in the range of 1 and 8
    }
    while (height < 1 || height > 8);

    // start loop on which # gets printed based on the provided height input of user
    for (int i = 0; i < height; i++)
    {
        // print left spacing as long as j < height whereas j is depending on i so the number of spacings decrease while i and the number of # increase
        for (int j = i + 1; j < height; j++)
        {
            printf(" ");
        }
        // prints pyramid in dependency to i
        for (int k = 0; k <= i; k++)
        {
            printf("#");
        }
        // starts new line after each row
        printf("\n");
    }
}

File: 07_week_06_python/04_sentimental-mario-more/mario.py
# import get_int function from cs50 library
from cs50 import get_int

# initialize while-loop
while True:

    # ask user for height of the pyramid
    height = get_int("Height: ")

    # if user input is not a positive integer no greater than 8, ask again
    if height not in range(1, 9):
        continue

    # if user input is in the right range
    if height in range(1, 9):

        # start loop on which # gets printed based on the provided height input of user
        for i in range(0, height, 1):

            # print left spacing as long as j < height whereas j is depending on i so the number of spacings decrease while i and the number of # increase
            for j in range(i + 1, height, 1):
                print(" ", end="")

            # prints left half of pyramid in dependency to i
            for k in range(0, i + 1, 1):
                print("#", end="")

            # print two spaces
            print("  ", end="")

            # prints right half of pyramid in dependency to i
            for l in range(0, i + 1, 1):
                print("#", end="")

            # starts new line after each row
            print("\n", end="")

    # break out of while-loop
    break